src/pipert2/core/base/flow.py

```diff
@@ -89,10 +89,18 @@ def execute_event(self, event: Method) -> None:
 
         """
 
-        for routine in self.routines.values():
-            routine.execute_event(event)
+        is_specific_flow_exist = True if event.flow_name else False
+        is_event_valid = (is_specific_flow_exist and event.flow_name == self.name) or True
```

Comment:
Ignoring the or True,
Maybe add this test to the Method class for checking if the flow and routine in the Method.